cancat/ccp/ccp_follower.py

```diff
@@ -0,0 +1,1422 @@
+        elif CCP_CRO_Type == CCP_GET_SEED:
+            parsed_msg = self._parse_get_seed_CRO(msg)
+        elif CCP_CRO_Type == CCP_BUILD_CHKSUM:
+            parsed_msg = self._parse_build_chksum_CRO(msg)
+        elif CCP_CRO_Type == CCP_UNLOCK:
+            parsed_msg = self._parse_unlock_CRO(msg)
+        elif CCP_CRO_Type == CCP_SET_S_STATUS:
+            parsed_msg = self._parse_set_s_status_CRO(msg)
+        elif CCP_CRO_Type == CCP_GET_S_STATUS:
+            parsed_msg = self._parse_get_s_status_CRO(msg)
+        else:
+            raise Exception("Cannot parse message type ", CCP_CRO_Type)
+
+        return parsed_msg
+
+    def _parse_connect_CRO(self, msg):
+        '''
+        CRO
+
+        +-------+--------+-----------------------------------------------+
+        |   0   |  byte  |  CRO command type                             |
+        +-------+--------+-----------------------------------------------+
+        |   1   |  bytes |  Counter (CTR)                                |
+        +-------+--------+-----------------------------------------------+
+        |  2..3 |  word  |  station address (Intel format)               |
+        +-------+--------+-----------------------------------------------+
+        |  4..7 |  bytes |  don't care                                   |
+        +-------+--------+-----------------------------------------------+
+        '''
+
+        cmd = utils._parse_byte(msg[0])
+        ctr = utils._parse_byte(msg[1])
+
+        stat_addr = utils._parse_2_byte_value(msg[2:4])
+
+        parsed = {'CMD': cmd, 'CTR': ctr, 'stat_addr': stat_addr}
+
+        return parsed
+
+    def _parse_exchangeID_CRO(self, msg):
+        '''
+        EXCHANGE_ID CRO
+
+        The CCP leader and follower stations exchange IDs for automatic session
+        configuration. This might include automatic assignment of a data
+        acquisition setup file depending on the follower's returned ID (Plug-n-Play).
+
+        +-------+--------+-----------------------------------------------+
+        |   0   |  byte  |  CRO command type                             |
+        +-------+--------+-----------------------------------------------+
+        |   1   |  bytes |  Counter (CTR)                                |
+        +-------+--------+-----------------------------------------------+
+        | 2..7  |  bytes |  CCP leader device ID information             |
+        |       |        |  (optional and implementation specific)       |
+        +-------+--------+-----------------------------------------------+
+        '''
+
+        cmd = utils._parse_byte(msg[0])
+        ctr = utils._parse_byte(msg[1])
+
+        # Device ID is implementation specific and optional, just return back value for now
+        id = utils._parse_6_byte_value(msg[2:])
+
+        parsed = {'CMD': cmd, 'CTR': ctr, 'DEVICE_ID': id}
+
+        return parsed
+
+    def _parse_set_MTA_CRO(self, msg):
```

Comment:
how is this used?  i'm not seeing it, aside from the test functions.  if it's part of an API, why is it named with a leading "_"?